### DISCUSSION TITLE
Asynchronous invocation / queue for serverless functions.

### DISCUSSION THREAD

**Discussion by ilievskizoran** [vercel/community#1348] [Help]
Hello all 👋 

I'm looking to use vercel to develop an application which receives and responds to a webhook and runs a separate function using the data.

The response to the webhook must go out within five seconds, so it is advised to defer the rest of the application logic till after the response.

I've been able to do this on AWS with [Asynchronous Lambda](https://docs.aws.amazon.com/lambda/latest/dg/invocation-async.html) invocations, where 1 lambda receives the hook, invokes another one async and responds to the hook under 5 seconds, but I cannot figure out a way of doing this with vercel serverless functions.

I also though about some sort of queue that might be used.

Any help appreciated on the issue

**Comment by gjunkie** [comment]
Were you able to find an answer?

**Reply by ilievskizoran** [reply]
Unfortunately not.

Vercel functions in all honesty are probably not the right tool for the type of system i was looking to have back then and there would be better alternatives.

All of the cloud providers would have products to handle the webhook response and message to a queue scenario, such as AWS Lambda to SQS or EventBridge.

**Reply by gjunkie** [reply]
Ah gotcha. Thanks for responding! I'm using MongoDB Atlas and trying to use its triggers to achieve some of this, but not sure how I'll handle auth yet.
